pkg/controller/data_download_controller.go

```diff
@@ -343,6 +357,8 @@ func (r *DataDownloadReconciler) Reconcile(ctx context.Context, req ctrl.Request
 			}
 		}
 
+		// Reservation is automatically completed in CreateMicroServiceBRWatcher
```

Comment:
Let's call `ReleaseReservation` explicitly, reasons:
1. more human readable
2. decouple of each function